File: app/src/main/java/com/cna/mineru/cna/Utils/DefaultInputDialog.java
package com.cna.mineru.cna.Utils;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.v4.app.DialogFragment;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.TextView;

import com.cna.mineru.cna.Adapter.DefaultInputAdapter;
import com.cna.mineru.cna.DB.UserSQLClass;
import com.cna.mineru.cna.DTO.ListViewBtnItem;
import com.cna.mineru.cna.R;

import java.util.ArrayList;
import java.util.Objects;

import static android.content.Context.MODE_PRIVATE;

public class DefaultInputDialog extends DialogFragment {
    private OnMyDialogResult mDialogResult;
    private ListView lv;
    private DefaultInputAdapter mAdapater;
    private int mode;

    @Override
    public Dialog onCreateDialog(Bundle saveInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        LayoutInflater inflater = Objects.requireNonNull(getActivity()).getLayoutInflater();

        View dig = inflater.inflate(R.layout.defaultinput_dialog, null);
        TextView tv_title = dig.findViewById(R.id.tv_title);
        TextView tv_detail = dig.findViewById(R.id.tv_detail);
        lv = dig.findViewById(R.id.listview1);
        TextView btn_cancel = (TextView) dig.findViewById(R.id.btn_cancel);

        UserSQLClass db = new UserSQLClass(this.getContext());

        SharedPreferences pref = getActivity().getSharedPreferences("isClassConfig", MODE_PRIVATE);
        mode = pref.getInt("isClassConfig", 0);
        ArrayList<ListViewBtnItem> items = new ArrayList<>();
        loadItemsFromDB(items);

        mAdapater = new DefaultInputAdapter(dig.getContext(), R.layout.item2, items) ;
        lv.setAdapter(mAdapater);

        btn_cancel.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                DefaultInputDialog.this.getDialog().cancel();
            }
        });

        lv.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView parent, View v, int position, long id) {
                if(mode==0){
                    switch (position){
                        case 0:
                            SharedPreferences.Editor editor1 = pref.edit();
                            editor1.putInt("isClassConfig", 1);
                            editor1.apply();
                            mDialogResult.finish(1);
                            break;
                        case 1:
                            SharedPreferences.Editor editor2 = pref.edit();
                            editor2.putInt("isClassConfig", 2);
                            editor2.apply();
                            mDialogResult.finish(2);
                            break;
                        case 2:
                            SharedPreferences.Editor editor3 = pref.edit();
                            editor3.putInt("isClassConfig", 3);
                            editor3.apply();
                            mDialogResult.finish(3);
                            break;
                    }
                    DefaultInputDialog.this.getDialog().cancel();
                }else if(mode==1){
                    switch (position){
                        case 0:
                            mDialogResult.finish(1);
                            db.setClassId(11, "초등학교 1학년");
                            DefaultInputDialog.this.getDialog().cancel();
                            break;
                        case 1:
                            db.setClassId(12, "초등학교 2학년");
                            DefaultInputDialog.this.getDialog().cancel();
                            break;
                        case 2:
                            db.setClassId(13, "초등학교 3학년");
                            DefaultInputDialog.this.getDialog().cancel();
                            break;
                        case 3:
                            db.setClassId(14, "초등학교 4학년");
                            DefaultInputDialog.this.getDialog().cancel();
                            break;
                        case 4:
                            db.setClassId(15, "초등학교 6학년");
                            DefaultInputDialog.this.getDialog().cancel();
                            break;
                        case 5:
                            db.setClassId(16, "초등학교 6학년");
                            DefaultInputDialog.this.getDialog().cancel();
                            break;
                    }
                    SharedPreferences.Editor editor = pref.edit();
                    editor.putInt("isClassConfig", 0);
                    editor.apply();
                    DefaultInputDialog.this.getDialog().cancel();
                }else if(mode==2){
                    switch (position){
                        case 0:
                            db.setClassId(21, "중학교 1학년");
                            DefaultInputDialog.this.getDialog().cancel();
                            break;
                        case 1:
                            db.setClassId(22, "중학교 2학년");
                            DefaultInputDialog.this.getDialog().cancel();
                            break;
                        case 2:
                            db.setClassId(23, "중학교 3학년");
                            DefaultInputDialog.this.getDialog().cancel();
                            break;
                    }
                    SharedPreferences.Editor editor = pref.edit();
                    editor.putInt("isClassConfig", 0);
                    editor.apply();
                    DefaultInputDialog.this.getDialog().cancel();
                }else if(mode==3){
                    switch (position){
                        case 0:
                            db.setClassId(31, "고등학교 1학년");
                            DefaultInputDialog.this.getDialog().cancel();
                            break;
                        case 1:
                            db.setClassId(32, "고등학교 2학년");
                            DefaultInputDialog.this.getDialog().cancel();
                            break;
                        case 2:
                            db.setClassId(33, "고등학교 3학년");
                            DefaultInputDialog.this.getDialog().cancel();
                            break;
                    }
                    SharedPreferences.Editor editor = pref.edit();
                    editor.putInt("isClassConfig", 0);
                    editor.apply();
                    DefaultInputDialog.this.getDialog().cancel();
                }
            }
        }) ;

        builder.setView(dig);
        return builder.create();
    }

    public boolean loadItemsFromDB(ArrayList<ListViewBtnItem> list) {
        ListViewBtnItem item ;

        if (list == null) {
            list = new ArrayList<>() ;
        }

        if(mode==0) {
            item = new ListViewBtnItem() ;
            item.setText("초등학생") ;
            list.add(item) ;
            item = new ListViewBtnItem() ;
            item.setText("중학생") ;
            list.add(item) ;
            item = new ListViewBtnItem() ;
            item.setText("고등학생") ;
            list.add(item) ;
        }else if(mode==1){
            item = new ListViewBtnItem() ;
            item.setText("초등학생 1학년") ;
            list.add(item) ;
            item = new ListViewBtnItem() ;
            item.setText("초등학생 2학년") ;
            list.add(item) ;
            item = new ListViewBtnItem() ;
            item.setText("초등학생 3학년") ;
            list.add(item) ;
            item = new ListViewBtnItem() ;
            item.setText("초등학생 4학년") ;
            list.add(item) ;
            item = new ListViewBtnItem() ;
            item.setText("초등학생 5학년") ;
            list.add(item) ;
            item = new ListViewBtnItem() ;
            item.setText("초등학생 6학년") ;
            list.add(item) ;
        }else if(mode==2){
            item = new ListViewBtnItem() ;
            item.setText("중학생 1학년") ;
            list.add(item) ;
            item = new ListViewBtnItem() ;
            item.setText("중학생 2학년") ;
            list.add(item) ;
            item = new ListViewBtnItem() ;
            item.setText("중학생 3학년") ;
            list.add(item) ;
        }else if(mode==3){
            item = new ListViewBtnItem() ;
            item.setText("고등학생 1학년") ;
            list.add(item) ;
            item = new ListViewBtnItem() ;
            item.setText("고등학생 2학년") ;
            list.add(item) ;
            item = new ListViewBtnItem() ;
            item.setText("고등학생 3학년") ;
            list.add(item) ;
        }
        return true ;
    }

    public void setDialogResult(OnMyDialogResult dialogResult) {
        mDialogResult = dialogResult;
    }

    public interface OnMyDialogResult {
        void finish(int _class);
    }
}

File: app/src/main/java/com/cna/mineru/cna/Utils/DefaultInputDialog.kt
package com.cna.mineru.cna.Utils

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import android.widget.AdapterView
import android.widget.ListView

import com.cna.mineru.cna.Adapter.DefaultInputAdapter
import com.cna.mineru.cna.DB.UserSQLClass
import com.cna.mineru.cna.DTO.ListViewBtnItem
import com.cna.mineru.cna.R

import java.util.ArrayList
import java.util.Objects

import android.content.Context.MODE_PRIVATE
import androidx.fragment.app.DialogFragment
import kotlinx.android.synthetic.main.defaultinput_dialog.*
class DefaultInputDialog : DialogFragment() {
    private var mDialogResult: OnMyDialogResult? = null
    private var lv: ListView? = null
    private var mAdapater: DefaultInputAdapter? = null
    private var mode: Int = 0

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(activity)
        val inflater = Objects.requireNonNull(activity)?.getLayoutInflater()

        val dig = inflater?.inflate(R.layout.defaultinput_dialog, null)

        val db = UserSQLClass(context)

        val pref = activity?.getSharedPreferences("isClassConfig", MODE_PRIVATE)
        if (pref != null) {
            mode = pref.getInt("isClassConfig", 0)
        }
        val items = ArrayList<ListViewBtnItem>()
        loadItemsFromDB(items)

        if (dig != null) {
            mAdapater = DefaultInputAdapter(dig.context, R.layout.default_input_item, items)
        }
        lv!!.adapter = mAdapater

        btn_cancel.setOnClickListener {
            val editor = pref?.edit()
            editor?.putInt("isClassConfig", 0)
            editor?.apply()
            this@DefaultInputDialog.dialog?.cancel()
        }

        lv!!.onItemClickListener = AdapterView.OnItemClickListener { parent, v, position, id ->
            if (mode == 0) {
                when (position) {
                    0 -> {
                        val editor1 = pref?.edit()
                        editor1?.putInt("isClassConfig", 1)
                        editor1?.apply()
                        mDialogResult!!.finish(1)
                    }
                    1 -> {
                        val editor2 = pref?.edit()
                        editor2?.putInt("isClassConfig", 2)
                        editor2?.apply()
                        mDialogResult!!.finish(2)
                    }
                    2 -> {
                        val editor3 = pref?.edit()
                        editor3?.putInt("isClassConfig", 3)
                        editor3?.apply()
                        mDialogResult!!.finish(3)
                    }
                }
                this@DefaultInputDialog.dialog?.cancel()
            } else if (mode == 1) {
                when (position) {
                    0 -> {
                        mDialogResult!!.finish(1)
                        db.classId(11)
                        this@DefaultInputDialog.dialog?.cancel()
                    }
                    1 -> {
                        db.classId(12)
                        this@DefaultInputDialog.dialog?.cancel()
                    }
                    2 -> {
                        db.classId(13)
                        this@DefaultInputDialog.dialog?.cancel()
                    }
                    3 -> {
                        db.classId(14)
                        this@DefaultInputDialog.dialog?.cancel()
                    }
                    4 -> {
                        db.classId(15)
                        this@DefaultInputDialog.dialog?.cancel()
                    }
                    5 -> {
                        db.classId(16)
                        this@DefaultInputDialog.dialog?.cancel()
                    }
                }
                val editor = pref?.edit()
                editor?.putInt("isClassConfig", 0)
                editor?.apply()
                this@DefaultInputDialog.dialog?.cancel()
            } else if (mode == 2) {
                when (position) {
                    0 -> {
                        db.classId(21)
                        this@DefaultInputDialog.dialog?.cancel()
                    }
                    1 -> {
                        db.classId(22)
                        this@DefaultInputDialog.dialog?.cancel()
                    }
                    2 -> {
                        db.classId(23)
                        this@DefaultInputDialog.dialog?.cancel()
                    }
                }
                val editor = pref?.edit()
                editor?.putInt("isClassConfig", 0)
                editor?.apply()
                this@DefaultInputDialog.dialog?.cancel()
            } else if (mode == 3) {
                when (position) {
                    0 -> {
                        db.classId(31)
                        this@DefaultInputDialog.dialog?.cancel()
                    }
                    1 -> {
                        db.classId(32)
                        this@DefaultInputDialog.dialog?.cancel()
                    }
                }//                        case 2:
                //                            db.setClassId(33);
                //                            DefaultInputDialog.this.getDialog().cancel();
                //                            break;
                val editor = pref?.edit()
                editor?.putInt("isClassConfig", 0)
                editor?.apply()
                this@DefaultInputDialog.dialog?.cancel()
            }
        }

        builder.setView(dig)
        return builder.create()
    }

    override fun onDestroy() {
        super.onDestroy()
        val pref = activity?.getSharedPreferences("isClassConfig", MODE_PRIVATE)
        if (pref != null) {
            mode = pref.getInt("isClassConfig", 0)
        }
        if (mode == 1 || mode == 2 || mode == 3) {
            val editor = pref?.edit()
            editor?.putInt("isClassConfig", 0)
            editor?.apply()
        }
    }

    fun loadItemsFromDB(list: ArrayList<ListViewBtnItem>?): Boolean {
        var list = list
        var item: ListViewBtnItem

        if (list == null) {
            list = ArrayList<ListViewBtnItem>()
        }

        if (mode == 0) {
            item = ListViewBtnItem()
            item.text("초등학생")
            list.add(item)
            item = ListViewBtnItem()
            item.text("중학생")
            list.add(item)
            item = ListViewBtnItem()
            item.text("고등학생")
            list.add(item)
        } else if (mode == 1) {
            item = ListViewBtnItem()
            item.text("초등학생 1학년")
            list.add(item)
            item = ListViewBtnItem()
            item.text("초등학생 2학년")
            list.add(item)
            item = ListViewBtnItem()
            item.text("초등학생 3학년")
            list.add(item)
            item = ListViewBtnItem()
            item.text("초등학생 4학년")
            list.add(item)
            item = ListViewBtnItem()
            item.text("초등학생 5학년")
            list.add(item)
            item = ListViewBtnItem()
            item.text("초등학생 6학년")
            list.add(item)
        } else if (mode == 2) {
            item = ListViewBtnItem()
            item.text("중학생 1학년")
            list.add(item)
            item = ListViewBtnItem()
            item.text("중학생 2학년")
            list.add(item)
            item = ListViewBtnItem()
            item.text("중학생 3학년")
            list.add(item)
        } else if (mode == 3) {
            item = ListViewBtnItem()
            item.text("고등학생 1학년")
            list.add(item)
            item = ListViewBtnItem()
            item.text("고등학생 2학년")
            list.add(item)
            //            item = new ListViewBtnItem() ;
            //            item.setText("고등학생 3학년") ;
            //            list.add(item) ;
        }
        return true
    }

    fun setDialogResult(dialogResult: OnMyDialogResult) {
        mDialogResult = dialogResult
    }

    interface OnMyDialogResult {
        fun finish(_class: Int)
    }
}

private operator fun Int.invoke(i: Int) {

}
